### PR TITLE
[Theming] Use themes in colors in Javascript

### PR DESCRIPTION
**Pull request by zachstence**
### Checklist

- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [x] I have added to the docs where applicable

~~- I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable~~
~~- For UI or styling changes, I have added a screenshot or gif showing before & after~~
_We will review UI changes in https://github.com/evidence-dev/evidence/pull/2628_

### CODE REVIEW

**Review by ItsMeBrianD — CHANGES_REQUESTED** [review]

File: packages/ui/core-components/src/lib/unsorted/ui/Tabs/TabDisplay.svelte [hunk 1]
```diff
@@ -1,53 +1,26 @@
...
-		return isHex(inputColor) || isRGB(inputColor) || isHSL(inputColor);
-	}
-
-	function addOpacityToColor(colorString) {
-		if (isHex(colorString)) {
-			return colorString + '1a';
-		} else if (isRGB(colorString) || isHSL(colorString)) {
-			return colorString.replace(/(\)|\s|$)/, ', 0.1$1');
-		}
-	}
```
[the remark this answers is not in the document]
Comment (zachstence) [reply packages/ui/core-components/src/lib/unsorted/ui/Tabs/TabDisplay.svelte:43]: No, the background color has opacity applied here via `chroma-js`

https://github.com/evidence-dev/evidence/blob/87d18a692c84d3649925f8fe5cbbe6f12c23b0d0/packages/ui/core-components/src/lib/unsorted/ui/Tabs/TabDisplay.svelte#L11

File: packages/ui/core-components/src/lib/unsorted/viz/map/_USMap.svelte [hunk 2]
```diff
@@ -79,54 +82,73 @@
...
+				'#e0f3db',
+				'#ccebc5',
+				'#a8ddb5',
+				'#7bccc4',
+				'#4eb3d3',
+				'#2b8cbe',
+				'#0868ac',
+				'#084081'
+			];
+		}
```
[the remark this answers is not in the document]
Comment (zachstence) [reply packages/ui/core-components/src/lib/unsorted/viz/map/_USMap.svelte:134]: I only changed indentation here, and I want to keep this PR as much theming-only as possible, so I'm not going to refactor any JS code

File: packages/ui/core-components/src/lib/unsorted/viz/map/_USMap.svelte [hunk 3]
```diff
@@ -79,54 +82,73 @@
...
+		if ($activeTheme === 'dark') {
+			console.log(
+				'before',
+				colorArray.map((color) => chroma(color).css())
+			);
```
[the remark this answers is not in the document]
Comment (zachstence) [reply packages/ui/core-components/src/lib/unsorted/viz/map/_USMap.svelte]: [4d69ac2](https://github.com/evidence-dev/evidence/pull/2655/commits/4d69ac2886cf79b22a3a3b008ffa250159254da8)

File: packages/ui/core-components/src/lib/unsorted/ui/Tabs/Tabs.svelte [hunk 4]
```diff
@@ -8,7 +8,7 @@
 	import TabDisplay from './TabDisplay.svelte';
 
 	export let id;
-	export let color = 'hsla(207, 65%, 39%, 1)';
+	export let color;
```
[the remark this answers is not in the document]
Comment (zachstence) [reply packages/ui/core-components/src/lib/unsorted/ui/Tabs/Tabs.svelte]: [75120cb](https://github.com/evidence-dev/evidence/pull/2655/commits/75120cbb38a8aac3a09e13b48fa79226e8906119)

File: packages/ui/core-components/src/lib/themes.js [hunk 5]
```diff
@@ -38,9 +41,11 @@ const createSystemThemeStore = () => {
 
 /**
  * @typedef ThemeStores
+ * TODO could use some help naming activeTheme / theme
```
[the remark this answers is not in the document]
Comment (zachstence) [reply packages/ui/core-components/src/lib/themes.js]: [87d18a6](https://github.com/evidence-dev/evidence/pull/2655/commits/87d18a692c84d3649925f8fe5cbbe6f12c23b0d0)

File: packages/lib/component-utilities/src/echarts.js [hunk 6]
```diff
@@ -176,11 +179,19 @@ export default (node, option) => {
...
 
-	const updateChart = (newOption) => {
-		option = newOption;
+	/** @param {Options} newOptions */
+	const updateChart = (newOptions) => {
+		if (newOptions.theme !== options.theme) {
+			chart.dispose();
+			chart = init(node, newOptions.theme, {
+				renderer: useSvg ? 'svg' : (newOptions.renderer ?? 'canvas')
+			});
```
[the remark this answers is not in the document]
Comment (zachstence) [reply packages/lib/component-utilities/src/echarts.js]: [f590998](https://github.com/evidence-dev/evidence/pull/2655/commits/f59099822da76b65b129eb7b6b5cdfc97b7bfbc0)

File: packages/lib/component-utilities/src/echarts.js [hunk 7]
```diff
@@ -1,39 +1,42 @@
...
 import * as chartWindowDebug from './chartWindowDebug';
 
 /**
  * @typedef {import("echarts").EChartsOption & {
  * 		dispatch?: ReturnType<typeof import("svelte").createEventDispatcher>;
  * 		showAllXAxisLabels?: boolean;
+ *    theme: 'light' | 'dark';
  * 	}
- * } ActionParams
+ * } Options
```
[the remark this answers is not in the document]
Comment (zachstence) [reply packages/lib/component-utilities/src/echarts.js]: [262ca2d](https://github.com/evidence-dev/evidence/pull/2655/commits/262ca2d1287c57bc8774d79b00a88f5c71270089)

File: packages/ui/core-components/jsconfig.json [hunk 8]
```diff
@@ -8,6 +8,7 @@
 		"sourceMap": true,
 		"moduleResolution": "NodeNext",
 		"strict": true,
-		"module": "NodeNext"
+		"module": "NodeNext",
+		"types": ["@evidence-dev/tailwind"]
```
[the remark this answers is not in the document]
Comment (zachstence) [reply packages/ui/core-components/jsconfig.json:12]: [9b2e673](https://github.com/evidence-dev/evidence/pull/2655/commits/9b2e6737c7b52572d01564186b54a4895170a18f)

**Review by ItsMeBrianD — APPROVED** [review]

**Pull request merged**